module/data/item/class.mjs

```diff
@@ -94,4 +78,55 @@ export default class ClassData extends SystemDataModel.mixin(ItemDescriptionTemp
       ability: ""
     };
   }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Migrate the class's saves & skills into TraitAdvancements.
+   * @param {object} source  The candidate source data from which the model will be constructed.
+   * @private
+   */
+  static _migrateTraitAdvancement(source) {
+    const system = source.system ??= {};
+    system.advancement ??= [];
+    if ( system.advancement.find(a => a.type === "Trait") ) return;
+    let needsMigration = false;
+
+    if ( system.saves?.length ) {
+      const savesData = {
+        type: "Trait",
+        configuration: {
+          grants: system.saves.map(t => `saves:${t}`)
+        }
+      };
+      savesData.value = {
+        chosen: savesData.configuration.grants
+      };
+      system.advancement.push(new TraitAdvancement(savesData).toObject());
+      delete system.saves;
+      needsMigration = true;
+    }
+
+    if ( system.skills?.choices?.length ) {
+      const skillsData = {
+        type: "Trait",
+        configuration: {
+          choices: [{
+            count: system.skills.number ?? 1,
+            pool: system.skills.choices.map(t => `skills:${t}`)
+          }]
+        }
+      };
+      if ( system.skills.value?.length ) {
+        skillsData.value = {
+          chosen: system.skills.value.map(t => `skills:${t}`)
+        };
+      }
+      system.advancement.push(new TraitAdvancement(skillsData).toObject());
+      delete system.skills;
+      needsMigration = true;
+    }
+
+    if ( needsMigration ) foundry.utils.setProperty(source, "flags.dnd5e.needsMigration", true);
```

Comment:
`needsMigration` seems a bit too general to me. Can we call it something like `persistSourceMigration` instead? That goes for the naming in `migration.mjs` as well. I'm hoping to make it clear what we're doing in those cases.